src/front/spv/mod.rs

```diff
@@ -2843,17 +2875,7 @@ impl<I: Iterator<Item = u32>> Parser<I> {
             });
 
             let binding = decor.io_binding().ok();
-            let offset = match decor.offset {
-                Some(offset) => offset,
-                None => match members.last() {
-                    Some(member) => {
-                        //TODO: is this needed?
-                        // If offsets are required, we can just put 0 here
-                        member.offset + module.types[member.ty].inner.span(&module.constants)
-                    }
-                    None => 0,
-                },
-            };
+            let offset = decor.offset.unwrap_or(0); //unexpected
```

Comment:
I think the SPIR-V spec language that covers this is:

> Composite objects in the StorageBuffer, PhysicalStorageBuffer, Uniform, and PushConstant Storage Classes must be explicitly laid out. The following apply to all the aggregate and matrix types describing such an object, recursively through their nested types:
> – Each structure-type member must have an Offset decoration.

It seems like this wouldn't apply to structs used for local variables, say. Does that matter here?

Reply:
yeah, I'm changing this to an error now. Just was lazy and didn't want to panic there.

Reply:
Actually, no, we still need to support the case where offsets are not present there: for interface structs (in the `Output` storage class, for example).